src/com/amazon/ion/benchmark/DataConstructor.java

```diff
@@ -224,12 +228,78 @@ public static IonValue constructIonData(ReparsedType parsedTypeDefinition) {
                     return SYSTEM.newBlob(constructLobs(constraintMapClone));
                 case CLOB:
                     return SYSTEM.newClob(constructLobs(constraintMapClone));
+                case STRUCT:
+                    return constructIonStruct(constraintMapClone);
+                case LIST:
+                    return constructIonList(constraintMapClone);
                 default:
                     throw new IllegalStateException(type + " is not supported.");
             }
         }
     }
 
+    /**
+     * Constructing IonStruct which is aligned with the constraints provided in the constraintMap.
+     * @param constraintMapClone collects the constraints from ISL file, the key represents the name of constraints,
+     * and the value is constraint value in ReparsedConstraint format.
+     * @return the constructed IonStruct value.
+     */
+    private static IonValue constructIonStruct(Map<String, ReparsedConstraint> constraintMapClone) {
```

Comment:
Should the return type be `IonStruct`?

Reply:
> Should the return type be `IonStruct`?

Yes, it would be more clear to return IonStruct.